lms/static/scripts/postmessage_json_rpc/server/methods-test.js

```diff
@@ -21,3 +21,25 @@ describe('postmessage_json_rpc/methods#requestConfig', () => {
     assert.deepEqual(requestConfig(), { foo: 'bar' });
   });
 });
+
+describe('postmessage_json_rpc/methods#requestGroups', () => {
```

Comment:
Jon, i don't want to put too much JS burden on you, so if you want to pass this off, I can refactor a bit.

We can restructure this test file like this and reuse a bit of the `beforeEach` stuff

```describe('postmessage_json_rpc/methods#requestConfig', () => {
  let configEl;

  beforeEach('inject the client config into the document', () => {
    configEl = document.createElement('script');
    configEl.setAttribute('type', 'application/json');
    configEl.classList.add('js-config');
    configEl.textContent = JSON.stringify({
      hypothesisClient: { foo: 'bar' },
    });
    document.body.appendChild(configEl);
  });

  afterEach('remove the client config from the document', () => {
    configEl.parentNode.removeChild(configEl);
  });

  describe('#requestConfig', () => {
    it(' the config returnsobject', () => {
      assert.deepEqual(requestConfig(), { foo: 'bar' });
    });
  });

  describe('#requestGroups', () => {
    it('returns the list of groups filter', async () => {
      configEl.textContent = JSON.stringify({
        hypothesisClient: { services: [{ groups: ['groupid1', 'groupid2'] }] },
      });
      assert.deepEqual(await requestGroups(), ['groupid1', 'groupid2']);
    });

    it.only('returns the list of groups filter in dev mode', async () => {
      const clock = sinon.useFakeTimers();

      configEl.textContent = JSON.stringify({
        dev: true,
        hypothesisClient: { services: [{ groups: ['groupid1', 'groupid2'] }] },
      });

      const promise = requestGroups();
      clock.tickAsync(4000);
      const groups = await promise;
      assert.deepEqual(groups, ['groupid1', 'groupid2']);
    });
  });
});```

I also added a test for the sleep() method you made. If we're gonna be REALLY picky, we would stub out the math.random too, but I think thats overkill for just a placeholder function anyway